nsq/async.py

```diff
@@ -241,7 +251,7 @@ def close(self):
     def _read_size(self, data):
         try:
             size = struct.unpack('>l', data)[0]
-            self.stream.read_bytes(size, self._read_body)
+            self._read_bytes(size, self._read_body)
```

Comment:
sorry, forgot about this PR :frowning:...

Since `_read_bytes` is now capturing the exception, doesn't that mean we're now going to miss calling the `self.close()` in the exception handler for this block?